### PR TITLE
Drop python 3.6 and 3.7

### PR DESCRIPTION
**Pull request by larrybradley**
This PR updates the Python minversion and min versions of other dependencies (numpy, astropy, matplotlib, etc.) to keep in line with astropy.

### CODE REVIEW

File: .readthedocs.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ build:
   image: latest
 
 python:
-  version: 3.7
+  version: 3.9
```
Comment (pllim) [inline .readthedocs.yml:7]: Last time I tried, RTD won't build in Python 3.9 and I had to fall back to 3.8. 🤷

**Comment by pllim** [comment]
I think the title is wrong and should be "Drop python 3.6 and 3.7".

Looks like you need to update the CI matrix too.

**Comment by larrybradley** [comment]
The CI failures are unrelated.  Will fix in a separate PR.

File: .readthedocs.yml (hunk 1 above)
Comment (larrybradley) [reply .readthedocs.yml:7]: Ugh, I just now saw this.  Let's see.  🤞

Comment (larrybradley) [reply .readthedocs.yml:7]: Indeed, 3.9 is not allowed:  "Invalid "python.version": expected one of (2, 2.7, 3, 3.5, 3.6, 3.7, 3.8, pypy3.5), got 3.9"

Comment (pllim) [reply .readthedocs.yml:7]: Maybe @astrojuanlu can tell us when RTD supports Python 3.9. 😸

Comment (astrojuanlu) [reply .readthedocs.yml:7]: It supports 3.9 and 3.10 since October! :) https://blog.readthedocs.com/new-build-specification/

**Pull request merged**